source/velocity_boundary_conditions/gplates.cc

```diff
@@ -191,26 +176,21 @@ namespace aspect
         AssertThrow(dn_theta - n_theta <= 1e-5,
                     ExcMessage("The velocity file has a grid structure that is not readable. Please refer to the manual for a proper grid structure."));
 
-        if ((delta_theta != 0.0) || (delta_phi != 0.0))
-          {
-            Assert((delta_theta - numbers::PI / (n_theta-1)) <= 1e-7,  ExcMessage("Resolution changed during boundary conditions. Interpolation is not working correctly."));
-            Assert((delta_phi - 2*numbers::PI / n_phi) <= 1e-7,  ExcMessage("Resolution changed during boundary conditions. Interpolation is not working correctly."));
-          }
-
         delta_theta =   numbers::PI / (n_theta-1);
         delta_phi   = 2*numbers::PI / n_phi;
 
-        // swap pointers to old and new values, we overwrite the old ones
-        // and the new ones become the old ones
-        std::swap (velocity_values, old_velocity_values);
-
-        (*velocity_values).reinit(n_theta,n_phi);
-        velocity_positions.reinit(n_theta,n_phi);
+        /**
+         * Two tables (one for the theta and one for the phi component)
+         * which contain the velocities at every point.
+         * velocity_values[0] is the table for the theta component, whereas
+         * velocity_values[1] is the table for the phi component.
+         */
+        std::vector<Table<2,double> > velocity_values(2,Table<2,double>(n_theta,n_phi));
```

Comment:
Is the initialization list a C++11 feature? If so we should use the `std::cxx11::array`.